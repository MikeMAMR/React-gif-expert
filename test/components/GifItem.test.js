import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en el componente GifItem', () => {
    
    const title = "Hola";
    const url = "http://localhost/a";

    test('debe de mostrar el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con la url y el alt', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });

});