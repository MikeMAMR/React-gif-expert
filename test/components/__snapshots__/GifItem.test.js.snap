// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem debe de mostrar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="http://localhost/a"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
